core/state_connector.go

```diff
@@ -187,10 +193,61 @@ func envAttestors(key string) []common.Address {
 	return attestors
 }
 
-func attestationSelector() []byte {
-	return []byte{0x29, 0xbe, 0x4d, 0xb2}
+// unused 'blockTime' might be used for hard forks in the future.
+func stateConnectorContract(chainID *big.Int, blockTime *big.Int) common.Address {
+	switch {
+	case chainID.Cmp(costonChainID) == 0:
+		return common.HexToAddress("0x947c76694491d3fD67a73688003c4d36C8780A97")
+	case chainID.Cmp(songbirdChainID) == 0:
+		return common.HexToAddress("0x3A1b3220527aBA427d1e13e4b4c48c31460B4d91")
+	case chainID.Cmp(flareChainID) == 0:
+		return common.HexToAddress("0x1000000000000000000000000000000000000001")
+	default:
+		return common.HexToAddress("0x1000000000000000000000000000000000000001")
+	}
+}
+
+func isStateConnectorActivated(chainID *big.Int, blockTime *big.Int) bool {
+	switch {
+	case chainID.Cmp(costonChainID) == 0:
+		return blockTime.Cmp(costonActivationTime) >= 0
+	case chainID.Cmp(songbirdChainID) == 0:
+		return blockTime.Cmp(songbirdActivationTime) >= 0
+	case chainID.Cmp(flareChainID) == 0:
+		return blockTime.Cmp(flareActivationTime) >= 0
+	default:
+		return true
+	}
+}
+
+// unused 'chainID' and 'blockTime' might be used for hard forks in the future.
+func submitAttestationSelectorFunc(chainID *big.Int, blockTime *big.Int) []byte {
+	switch {
+	default:
+		return []byte{0xcf, 0xd1, 0xfd, 0xad}
+	}
+}
```

Comment:
Same here, for now I'd rather have it just return `[]byte{0xcf, 0xd1, 0xfd, 0xad}` without the switch or unused parameters, but it's up to you.